src/models/category.rs

```diff
@@ -93,43 +93,42 @@ impl Category {
         limit: i64,
         offset: i64,
     ) -> QueryResult<Vec<Category>> {
-        use diesel::select;
-        use diesel::dsl::*;
-
         let sort_sql = match sort {
             "crates" => "ORDER BY crates_cnt DESC",
             _ => "ORDER BY category ASC",
         };
 
         // Collect all the top-level categories and sum up the crates_cnt of
         // the crates in all subcategories
-        select(sql::<categories::SqlType>(&format!(
-            "c.id, c.category, c.slug, c.description,
-                sum(c2.crates_cnt)::int as crates_cnt, c.created_at
-             FROM categories as c
-             INNER JOIN categories c2 ON split_part(c2.slug, '::', 1) = c.slug
-             WHERE split_part(c.slug, '::', 1) = c.slug
-             GROUP BY c.id
+        sql_query(format!(
+            "SELECT c.id, c.category, c.slug, c.description, \
```

Comment:
How would you feel about moving this into a separate file and `include_str!`ing it. We can also change this to call `.bind` rather than using string interpolation (other than the order by)

Reply:
:+1: